docs/src/rust-client/delegated_proving_tutorial.md

```diff
@@ -0,0 +1,195 @@
+
+## What we'll cover
+
+- Explaining what "delegated proving" is and its pros and cons
+- How to use delegated proving with the Rust client
+
+## What is Delegated Proving?
+
+Before diving into our code example, let's clarify what in the world "delegated proving" actually is.
+
+Delegated proving is the process of outsourcing a part of the ZK proof generation of your transaction to a third party. For certain computationally constrained devices such as mobile phones and web browser environments, generating ZK proofs might take too long to ensure an acceptable user experience. Devices that do not have the computational resources to generate Miden proofs in under 1-2 seconds can use delegated proving to provide a more responsive user experience.
+
+_How does it work?_ When a user choses to use delegated proving, they send off a portion of the zk proof of their transaction to a dedicated server. This dedicated server generates the remainder of the ZK proof of the transaction and submits it to the network. Submitting a transaction with delegated proving is trustless, meaning if the delegated prover is malicious, the could not compromise the security of the account that is submitting a transaction to be processed by the delegated prover. The only downside of using delegated proving is that it reduces the privacy of the account that uses delegated proving, because the delegated prover would have knowledge of the transaction that is being proven.
+
+Anyone can run their own delegated prover server. If you are building a product on Miden, it may make sense to run your own delegated prover server for your users. To run your own delegated proving server, follow the instructions here: https://crates.io/crates/miden-proving-service
+
+## Step 1: Initialize your repository
+
+Create a new Rust repository for your Miden project and navigate to it with the following command:
+
+```bash
+cargo new miden-delegated-proving-app
+cd miden-delegated-proving-app
+```
+
+Add the following dependencies to your `Cargo.toml` file:
+
+```toml
+[dependencies]
+miden-client = { version = "0.8.1", features = ["testing", "concurrent", "tonic", "sqlite"] }
+miden-lib = { version = "0.8", default-features = false }
+miden-objects = { version = "0.8", default-features = false }
+miden-crypto = { version = "0.14.0", features = ["executable"] }
+miden-assembly = "0.14.0"
+rand = { version = "0.9" }
+serde = { version = "1", features = ["derive"] }
+serde_json = { version = "1.0", features = ["raw_value"] }
+tokio = { version = "1.40", features = ["rt-multi-thread", "net", "macros"] }
+rand_chacha = "0.9.0"
+miden-client-tools = "0.1.1"
+
+```
+
+## Step 2: Initialize the client and delegated prover endpoint and construct transactions
+
+Similarly to previous tutorials, we must instantiate the client.
+We construct a `RemoteTransactionProver` that points to our delegated-proving service running at http://0.0.0.0:8082.
```

Comment:
Can you say in two sentences what you do in this tutorial? Create 10 P2ID notes and prove remotely or so